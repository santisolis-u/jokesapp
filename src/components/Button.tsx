import {
  Pressable,
  StyleProp,
  StyleSheet,
  Text,
  View,
  ViewStyle,
} from 'react-native';
import React from 'react';

interface ButtonProps {
  children: JSX.Element;
  style: StyleProp<ViewStyle>;
}

const Button: React.FC<ButtonProps> = ({children, style}) => {
  return (
    <View style={[style, styles.buttonContainer]}>
      <Pressable style={styles.pressable}>
        <Text style={styles.text}>{children}</Text>
      </Pressable>
    </View>
  );
};

export default Button;

const styles = StyleSheet.create({
  buttonContainer: {
    marginVertical: 40,
    justifyContent: 'center',
    alignItems: 'center',
    height: 60,
    backgroundColor: 'black',
    width: '80%',
  },
  pressable: {},
  text: {
    fontFamily: 'OpenSans-Regular',
    color: 'white',
  },
});
