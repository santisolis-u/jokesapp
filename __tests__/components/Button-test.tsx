import 'react-native';
import React from 'react';
import Button from '../../src/components/Button';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Button', () => {
  it('renders correctly', () => {
    const json = renderer.create(<Button title={'TEST'} />).toJSON();
    expect(json).toMatchSnapshot();
  });
});
