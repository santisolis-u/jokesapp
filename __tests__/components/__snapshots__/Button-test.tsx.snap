// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "height": 60,
      "justifyContent": "center",
      "marginVertical": 40,
      "padding": 20,
      "width": "80%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "OpenSans-Regular",
      }
    }
  >
    TEST
  </Text>
</View>
`;
